huaweicloud/services/cci/resource_huaweicloud_cci_network.go

```diff
@@ -0,0 +1,269 @@
+							"and end with a character or digit."),
+					validation.StringLenBetween(1, 200),
+				),
+			},
+			"security_group_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"network_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cidr": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceNetworkAnnotations(d *schema.ResourceData, config *config.Config) map[string]string {
+	result := map[string]string{
+		"network.alpha.kubernetes.io/domain_id":  config.DomainID,
+		"network.alpha.kubernetes.io/project_id": config.HwClient.ProjectID,
+	}
+	if v, ok := d.GetOk("security_group_id"); ok {
+		result["network.alpha.kubernetes.io/default-security-group"] = v.(string)
+	}
+	return result
+}
+
+func resourceCciNetworkCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	cciClient, err := config.CciV1BetaClient(region)
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud CCI Beta v1 client: %s", err)
+	}
+
+	networkId := d.Get("network_id").(string)
+	subnet, err := vpc.GetVpcSubnetById(config, region, networkId)
+	if err != nil {
+		return fmtp.DiagErrorf("The subnet does not exist: %s", err)
+	}
+
+	opt := networks.CreateOpts{
+		Kind:       "Network",
+		ApiVersion: "networking.cci.io/v1beta1",
+		Metadata: networks.CreateMetaData{
+			Name:        d.Get("name").(string),
+			Annotations: resourceNetworkAnnotations(d, config),
+		},
+		Spec: networks.Spec{
+			AvailableZone: d.Get("availability_zone").(string),
+			NetworkType:   "underlay_neutron",
+			AttachedVPC:   subnet.VPC_ID,
+			NetworkID:     networkId,
+			SubnetID:      d.Get("subnet_id").(string),
+			Cidr:          d.Get("cidr").(string),
+		},
+	}
+
+	ns := d.Get("namespace").(string)
+	create, err := networks.Create(cciClient, ns, opt).Extract()
+
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud CCI Network: %s", err)
+	}
+
+	logp.Printf("[DEBUG] Waiting for HuaweiCloud CCI network (%s) to become available", create.Metadata.Name)
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"Initializing", "Pending"},
+		Target:       []string{"Active", "Failed"},
+		Refresh:      waitForCciNetworkActive(cciClient, ns, create.Metadata.Name),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmtp.DiagErrorf("Error obtain HuaweiCloud CCI network status: %s", err)
+	}
+	d.SetId(create.Metadata.Name)
```

Comment:
after the resource is created, it is handed over to Terraform, so the `ID` should be set before waitfor

Reply:
The `Failed` state has been removed, and the ID is moved to the front of the waitfor method.